nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -906,6 +1113,22 @@ impl BackgroundTask for SupportBundleCollector {
     }
 }
 
+async fn write_ereport(ereport: Ereport, dir: &Utf8Path) -> anyhow::Result<()> {
+    let sn =
+        ereport.metadata.serial_number.as_deref().unwrap_or("unknown_serial");
+    let dir = dir.join(sn).join(ereport.id.restart_id.to_string());
+    tokio::fs::create_dir_all(&dir)
+        .await
+        .with_context(|| format!("failed to create directory '{dir}'"))?;
+    let file_path = dir.join(format!("{}.json", ereport.id.ena));
```

Comment:
Just to confirm, the schema being used is:

```
{serial_number}/{restart_id}/{ena}.json
```

(Might be nice to summarize that somewhere)

Also, how confident are we all these values will always be filesystem-safe (e.g., not containing null characters, slashes, etc?)

Reply:
Yeah, that's correct --- I mentioned it in the PR description but there probably ought to be a comment. 

As for the other question, they ought to be: `restart_id` is always a UUID, and `ena` is always an integer. The serial is the only thing that Nexus doesn't have the ability to validate, but the SP really should not emit a string that isn't safe to use in a filename AFAICT.

Reply:
The "easy way out" would be to just store these in directories for each restart ID, but that seems unfortunate --- it seems to me that if i'm investigating a support bundle, I'd want to be able to see all the ereports from a particular device, across restarts.

Reply:
Yeah, I agree that organizing by device seems reasonable. I just flagged it as "arbitrary input we're treating as a string" - I'd rather throw an error in that case, instead of walking the filesystem arbitrarily, but I think this schema will work for most non-hostile cases

Reply:
Opened #8742 for serials. It would be nice to be able to share code with the Hubris crate, potentially.

Reply:
Great, this seems like plenty validation for now - thanks for doing this!